### PR TITLE
Add Organizations, OrganizationMembership, and Identities to the library

### PR DESCRIPTION
**Pull request by doomspork**
I'm targeting our fork's master since I figure we'll want to make some additional changes and clean up the tests a bit before we open a PR upstream. Need to learn how the `spec` library to use works, I'm only familiar with ExUnit unfortunately.

Thoughts @btkostner?

### CODE REVIEW

File: lib/zen_ex/core/entities/ticket.ex [hunk 1]
```diff
@@ -1,9 +1,35 @@
 defmodule ZenEx.Entity.Ticket do
-
-  defstruct [:id, :url, :external_id, :created_at, :updated_at, :type, :subject, :raw_subject,
-             :description, :priority, :status, :recipient, :requester_id, :submitter_id, :assignee_id,
-             :organization_id, :group_id, :collaborator_ids, :forum_topic_id, :problem_id, :has_incidents,
-             :due_at, :tags, :via, :ticket_form_id, :custom_fields, :satisfaction_rating, :sharing_agreement_ids]
+  defstruct [
```
Comment (btkostner) [inline lib/zen_ex/core/entities/ticket.ex:2]: nitpick but the moduledoc should go first

**Review by btkostner — APPROVED** [review]
Looks good. Lots of lint fixes too 👍🏻

File: lib/zen_ex/core/entities/ticket.ex (hunk 1 above)
Comment (doomspork) [reply lib/zen_ex/core/entities/ticket.ex:2]: This how they've done it in their library but I'm up for making changes that reflect the community styleguide :+1:

Comment (btkostner) [reply lib/zen_ex/core/entities/ticket.ex:2]: Baby steps. I'd probably work on the code first and then we can clean it up once we get some cred from the author 

:parrotwave1: :parrotwave2: :parrotwave3: :parrotwave4: :parrotwave5: :parrotwave6: :parrotwave7: :parrotwave8: :parrotwave9:

Comment (doomspork) [reply lib/zen_ex/core/entities/ticket.ex:2]: Yup! I'm trying to minimize the changes (many of these are the result of the formatter running) and focus on functionality for our needs. We can iterate on the overall code quality and test coverage.

**Pull request merged**